snooty/parser.py

```diff
@@ -433,6 +435,34 @@ def dispatch_departure(self, node: docutils.nodes.Node) -> None:
                     repr(popped),
                 )
 
+        if (
+            isinstance(popped, n.Directive)
+            and f"{popped.domain}:{popped.name}" == ":glossary"
+        ):
+
+            definition_list = next(popped.get_child_of_type(n.DefinitionList), None)
+
+            if definition_list is None:
```

Comment:
Our tests should take this branch.

It might also be worth handling the case where there's more than one child node? if len(popped.children) != 1

Reply:
if a glossary has more than one child node - does that mean it has two separate definition lists children? or it has two or more `definition_list_item` children without a `definition_list parent container`?